docs/naming_conventions.md

```diff
@@ -0,0 +1,151 @@
+   (e.g. `gender`, `birthDate`) for resource attributes
+
+   Other FHIR models have used one of or a mix of the following:
+
+   - kebab-case (e.g. `age-at-event`, `sequencing-experiment`)
+   - PascalCase (see above)
+   - camelCase (see above)
+
+## Resource File Name
+
+### Conformance Resources
+
+All conformance resources should follow this file naming pattern:
+
+**Pattern**
+```
+<resource type>_<base definition resource id>_<resource id>.json
+```
+
+**Example StructureDefinition Resource**
+```json
+{
+  "resourceType": "StructureDefinition",
+  "id": "participant",
+  "url": "http://fhir.kids-first.io/StructureDefinition/participant",
+  "name": "Participant",
+  "status": "draft",
+  "fhirVersion": "4.0.0",
+  "kind": "resource",
+  "abstract": false,
+  "type": "Patient",
+  "baseDefinition": "http://hl7.org/fhir/StructureDefinition/Patient",
+  "derivation": "constraint",
+  "differential": {
+    "element": [
+      {
+        "id": "Patient.gender",
+        "path": "Patient.gender",
+        "min": 1
+      },
+      {
+        "id": "Patient.birthDate",
+        "path": "Patient.birthDate",
+        "max": "0"
+      }
+    ]
+  }
+}
+```
+
+**File Name Example**
+```
+StructureDefinition_Patient_participant.json
+```
+
+**Additional Examples**
+```
+# Extension type StructureDefinition
+StructureDefinition_Patient_age-at-event.json
+```
+
+### Non-Conformance Resources
+
+All non-conformance resources should follow this file naming pattern:
+
+**NOTE:** Non-conformance resources have to follow a different pattern since
+they do not have a `baseDefinition` attribute. The rough equivalent would be
+the `meta.profile` element.
+
+**Pattern**
+```
+<resource type>_<profile id>_<resource id>.json
+```
+
+* Not sure how to handle a resource with multiple profiles
+
+**Example Patient**
+```json
+{
+    "resourceType":"Patient",
+    "id": "kf-participant",
+    "meta": {
+        "profile": [
+            "http://fhir.kids-first.io/StructureDefinition/participant"
+        ]
+    },
+    "gender": "female",
+    "name": [
+        {
+            "family":"Singh"
+        }
+    ]
+}
+```
+
+**File Name Example**
+```
+Patient_participant_kf-participant.json
+```
+
+### File Name Delimiter
+
+The file name delimiter will be an underscore `_`, since it is not contained
+in the character set for the resource `id`.
+
+## Other Things to Keep in Mind
+
+### SearchParameter Generation
+
+We will autogenerate search parameters off of our extensions. The `id`
+will be used as the `SearchParameter.code` (the term use use in the
+query part of the URL).
+
+Depending on which casing we choose for `id` values, we may want to
+to convert something like `AgeAtEvent` to `age-at-event` before using
+it as the `SearchParameter.code` so that a query string in the URL
+would look like this:
+
+```
+/Observation?_profile=<Phenotype url>&age-at-event=10
+```
+
+instead of this:
+```
+/Observation?_profile=<Phenotype url>&AgeAtEvent=10
+```
```

Comment:
But if we do this conversion then there's a secret disconnect between what you see and what you need to search with.

Reply:
I think that's the point I'm making. We probably don't want to do the secret conversion, so we have to be ok with how our chosen `id`s look in URL query strings. I think I'd like to go with kebab-case for this reason and because I personally think its more readable than various forms of camel case.